### PR TITLE
database implementation.

### PR DESCRIPTION
**Pull request by cam28**
[AB#77](https://dev.azure.com/dhk2/65e21a49-896d-470d-8659-fc7cca2f4625/_workitems/edit/77)

### CODE REVIEW

File: src/DatabaseMain.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import src.InitializeDatabaseScript as DB
+
+
+if __name__ == "__main__":
+    db = DB.Database()
+    user_name = "postgres"
+    password = "my_secret_password"
+    host = "localhost"
+    port = 5432
+    database_name = "test_db5"
```
Comment (DanaKhoshnaw) [inline src/DatabaseMain.py:10]: We may be able to reference the variables used in the ini file for these values

File: src/DatabaseMain.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    table_name = "bug_table"
+    record = ("bug_title", "bug_description", "bug_traceback_info", False)
+    db.connect()
+    db.create_database()
```
Comment (DanaKhoshnaw) [inline src/DatabaseMain.py:14]: We should move this connect and create to the main file, where we do first when building the application

File: src/DatabaseMain.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    db.create_table()
```
Comment (DanaKhoshnaw) [inline src/DatabaseMain.py:15]: This as well

File: src/InitializeDatabaseScript.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            # Print PostgreSQL Connection properties
+            print(self.db_server_conn.get_dsn_parameters(), "\n")
+
+            # Print PostgreSQL version
+            # possibly do not need class based cursor so this might need to be removed.
+            # TODO
+
+            self.db_cursor.execute("SELECT version();")
+            record = self.db_cursor.fetchone()
+            print("You are connected to - ", record, "\n")
```
Comment (DanaKhoshnaw) [inline src/InitializeDatabaseScript.py:35]: We should remove any and all print statement

File: src/InitializeDatabaseScript.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+            cursor = conn.cursor()
+            # self.bug_table_name =
+
+            # basic Table
+            table = ("CREATE TABLE " + self.bug_table_name + """ (
+                        bug_title text NULL,
+                        bug_description text NULL,
+                        bug_traceback_info text NULL,
+                        bug_resolved BOOL NULL,
+                        PRIMARY KEY (bug_title)
```
Comment (DanaKhoshnaw) [inline src/InitializeDatabaseScript.py:83]: I think its safe to remove "bug" within all of these names and having bug in just the title of the table should be sufficient

File: src/InitializeDatabaseScript.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+                # sql_values.append(item)
+            # insert_values = (bug_record[0], bug, "traceback info test", False)
+            insert_query = (" INSERT INTO " + self.bug_table_name + """(
+                            bug_title, 
+                            bug_description, 
+                            bug_traceback_info, 
+                            bug_resolved) 
+                            VALUES (%s, %s, %s, %s) """)
+
+            cursor.execute(insert_query, bug_record)
```
Comment (DanaKhoshnaw) [inline src/InitializeDatabaseScript.py:116]: Will this insert every item in the bug_record object into the correct column?

File: src/DatabaseMain.py (hunk 2 above)
Comment (cam28) [reply src/DatabaseMain.py:14]: Done

File: src/DatabaseMain.py (hunk 1 above)
Comment (cam28) [reply src/DatabaseMain.py:10]: Done

File: src/DatabaseMain.py (hunk 3 above)
Comment (cam28) [reply src/DatabaseMain.py:15]: Done

File: src/InitializeDatabaseScript.py (hunk 5 above)
Comment (cam28) [reply src/InitializeDatabaseScript.py:83]: Done

File: src/InitializeDatabaseScript.py (hunk 6 above)
Comment (cam28) [reply src/InitializeDatabaseScript.py:116]: Yeah it does. But it depends on the order they are passed in as an argument in the bug_record and the order they are listed in the query

File: src/InitializeDatabaseScript.py (hunk 4 above)
Comment (cam28) [reply src/InitializeDatabaseScript.py:35]: Removed

**Pull request merged**